packages/flutter/lib/src/material_animated_icons/animated_icons.dart

```diff
@@ -81,7 +82,164 @@ class AnimatedIcon extends StatelessWidget {
 
   @override
   Widget build(BuildContext context) {
-    // TODO(amirh): implement this.
-    return new Container();
+    // TODO(amirh): implement semantics, text direction, scaling.
+    final _AnimatedIconData iconData = icon;
+    return new CustomPaint(
+      size: iconData.size,
+      painter: new _AnimatedIconPainter(
+        iconData.paths,
+        progress,
+        color,
+        () => new ui.Path(),
+      ),
+    );
   }
 }
+
+typedef ui.Path _UiPathFactory();
+
+class _AnimatedIconPainter extends ChangeNotifier implements CustomPainter {
+  _AnimatedIconPainter(
+    this.paths,
+    this.progress,
+    this.color,
+    this.uiPathFactory
+  ) {
+    progress.addListener(notifyListeners);
```

Comment:
this doesn't seem to have a matching removeListener

Reply:
Following f2f - switched to extending CustomPainter and passing the animation as the repaint listener